scaling.md

```diff
@@ -0,0 +1,78 @@
+# Scaling Application Service Adapter
+
+This topic provides guidance on scaling Application Service Adapter components
+for performance or availability.
+
+* [Notes on system performance](#system-performance)
+* [Scaling the Application Service Adapter controller managers](#controllers)
+* [Scaling the Application Service Adapter API](#api)
+
+## <a id="system-performance"></a>Notes on system performance
+
+The Application Service Adapter runs on a Kubernetes cluster and makes
+extensive use of the Kubernetes API to store and manage resources. For these
+reasons, the performance of the Application Service Adapter is very dependent on
+the underlying Kubernetes control plane. If you observe slow performance or
+timeouts when using the Application Service Adapter, a good first step is to
+ensure that the control plane for your Kubernetes cluster is properly resourced.
+
+For managed clusters that come from public-cloud IaaSes, you generally don't
+need to be concerned about scaling the control plane, but if you are managing
+your own Kubernetes clusters, you will need to ensure that you have provisioned
+adequate capacity for the `apiserver` and `etcd` components. See the [Kubernetes
+documentation](https://kubernetes.io/docs/setup/production-environment/) as a
+starting place for understanding production cluster concerns.
+
+## <a id="controllers"></a>Scaling the Application Service Adapter controller managers
+
+### Vertical Scaling
+The Application Service Adapter deploys a set of controller managers responsible
+for reconciling various custom resources into running pods on your Kubernetes
+cluster. If these controller managers are resource constrained, scaling them up
+vertically may improve performance.
+
+Controller manager deployments in the `tas-adapter-system` namespace are:
```

Comment:
I think we should explicitly call out that these controller maintain an in-memory "informer cache" of the events they are tracking and will  definitely require larger memory limits on large/very active environments.

Reply:
I took a crack at this. Let me know if it looks OK to you.

Reply:
yeah looks good 👍